### PR TITLE
Improve the trim incompatibility message for Assembly.GetType

### PR DESCRIPTION
**Pull request by vitek-karas**
Currently calling `Assembly.GetType` is always a warning when trimming. But typically the code can be changed to use `Type.GetType` with assembly qualified type name, which is a trim compatible pattern.

This modifies the warning message on the `RequiresUnreferencedCode` attribute for `Assembly.GetType` to suggest using `Type.GetType` instead.

### CODE REVIEW

**Comment by eerhardt** [comment]
Don't you need to update the `ref` assembly code as well?

**Comment by vitek-karas** [comment]
> Don't you need to update the ref assembly code as well?

Of course - I was... dumb. Should be fixed now.

File: src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs [hunk 1]
```diff
@@ -242,7 +242,7 @@ public override AssemblyName GetName(bool copiedName)
             return type;
         }
 
-        [RequiresUnreferencedCode("Types might be removed")]
+        [RequiresUnreferencedCode("Types might be removed by trimming. Consider using Type.GetType instead.")]
```
Comment (MichalStrehovsky) [inline src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs:245]: If we're updating CoreCLR's RuntimeAssembly, should we also update Mono's and NativeAOT's?

**Review by MichalStrehovsky — COMMENTED** [review]
If we want to be prescriptive in the message, should we change this to `Types might be removed by trimming. If the type name is a string literal, consider using Type.GetType instead.`.

Basically the advice to use `Type.GetType` is not useful if this is not a string constant that we can analyze or is properly annotated.

File: src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs (hunk 1 above)
Comment (vitek-karas) [reply src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs]: Done - thanks for reminding me

**Comment by vitek-karas** [comment]
Thanks for the suggestion @MichalStrehovsky  - I changed the message everywhere.

**Review by MichalStrehovsky — APPROVED** [review]
Thank you!

**Pull request merged**